osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/security/providers/HsmEncryptionProvider.java

```diff
@@ -45,63 +45,77 @@
 
     private final KeyStore keyStore;
 
-    public HsmEncryptionProvider(File keyStoreFile) {
+    public HsmEncryptionProvider(final File keyStoreFile) throws EncrypterException {
         try {
             super.setKeyFile(keyStoreFile);
             this.keyStore = KeyStore.getInstance(TYPE, PROVIDER);
-            FileInputStream fIn = new FileInputStream(keyStoreFile);
+            final FileInputStream fIn = new FileInputStream(keyStoreFile);
             this.keyStore.load(fIn, null);
-        } catch (CertificateException | NoSuchAlgorithmException | NoSuchProviderException | IOException | KeyStoreException e) {
+        } catch (final CertificateException | NoSuchAlgorithmException | NoSuchProviderException | IOException | KeyStoreException e) {
             throw new EncrypterException("Could not read keystore", e);
         }
     }
 
-    public Secret decrypt(EncryptedSecret secret, String keyReference) {
+    @Override
+    public byte[] decrypt(final EncryptedSecret secret, final String keyReference) throws EncrypterException {
 
-        Secret decryptedSecret = super.decrypt(secret, keyReference);
+        byte[] decryptedSecret = super.decrypt(secret, keyReference);
 
-        byte[] decryptedSecretBytes = decryptedSecret.getSecret();
+        if (decryptedSecret.length > KEY_LENGTH) {
 
-        if (decryptedSecretBytes.length > KEY_LENGTH) {
+            final byte[] truncatedDecryptedSecretBytes = Arrays.copyOfRange(decryptedSecret, 0,
+                    decryptedSecret.length-16);
 
-            byte[] truncatedDecryptedSecretBytes = Arrays.copyOfRange(decryptedSecretBytes, 0,
-                    decryptedSecretBytes.length-16);
-
-            LOGGER.trace("Truncating decrypted key from " + Hex.encodeHexString(decryptedSecretBytes) + " to " +
+            LOGGER.trace("Truncating decrypted key from " + Hex.encodeHexString(decryptedSecret) + " to " +
                             Hex.encodeHexString(truncatedDecryptedSecretBytes));
 
-            decryptedSecret = new Secret(truncatedDecryptedSecretBytes);
+            return truncatedDecryptedSecretBytes;
         }
 
         return decryptedSecret;
     }
 
-    protected Cipher getCipher() {
+    @Override
+    public byte[] generateAes128BitsSecret(String keyReference) throws EncrypterException {
+        //byte[] newSecret = new byte[128/8];
+        try {
+            //SecureRandom.getInstance("SHA1PRNG",PROVIDER).nextBytes(newSecret);
+            return this.encrypt(KeyGenerator.getInstance("AES").generateKey().getEncoded(),keyReference).getSecret();
+        } catch (NoSuchAlgorithmException exc) {
+            throw new EncrypterException("Could not generate secret", exc);
+        }
+        //return newSecret;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)